### PR TITLE
chore: add `tailcall-prettier`

### LINKED ISSUE #1730: chore: Create a `tailcall-prettier`
We use [prettier](https://prettier.io/) to format all our GraphQL, md, JSON & YAML files. Prettier is a purely Node.JS-based implementation that's opinionated about how text is formatted. 

**Requirements**
- Create a new workspace called `tailcall-prettier`
- Tailcall prettier provides a Rust based programmable API that takes in content and formats it using Prettier.
- Read files in rust and pipe them to Prettier. For eg:
    
    ```rust
    use std::process::{Command, Stdio};
    use std::io::Write;
    
    fn format_with_prettier(code: &str) -> Result<String, std::io::Error> {
        let mut child = Command::new("node")
            .arg("path_to_prettier_script.js")
            .stdin(Stdio::piped())
            .stdout(Stdio::piped())
            .spawn()?;
    
        if let Some(ref mut stdin) = child.stdin {
            stdin.write_all(code.as_bytes())?;
        }
    
        let output = child.wait_with_output()?;
        if output.status.success() {
            Ok(String::from_utf8_lossy(&output.stdout).to_string())
        } else {
            Err(std::io::Error::new(std::io::ErrorKind::Other, "Prettier formatting failed"))
        }
    }
    ```
- Ensure the local version of prettier is used.
- Execution should be fast and multithreaded if required.

### PR DESCRIPTION
**Pull request by ssddOnTop**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #1730
/claim 1730

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **New Features**
    - Introduced code formatting functionality using Prettier for specified file types.
    - Added a parser module to detect file types based on extensions.
    - Implemented asynchronous formatting using Tokio for improved performance.
- **Tests**
    - Included a test for formatting JavaScript code.
- **Chores**
    - Updated CI workflow to install Node.js, Prettier, and configure npm global path.

### CODE REVIEW

File: tailcall-prettier/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            "json" => Ok(FileTypes::Json),
+            "md" => Ok(FileTypes::Md),
+            "ts" => Ok(FileTypes::Ts),
+            "js" => Ok(FileTypes::Js),
+            _ => Err(anyhow!("Unsupported file type")),
+        }
+    }
+}
+
+pub fn format_with_prettier<T: AsRef<str>>(code: T, file_ty: T) -> Result<String> {
```
Comment (tusharmath) [inline tailcall-prettier/src/lib.rs:48]: Keep the API close to — 
https://github.com/prettier/prettier/blob/main/src/index.js#L62C29-L62C36

File: tailcall-prettier/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        match self {
+            FileTypes::Gql => write!(f, "graphql"),
+            FileTypes::Yml => write!(f, "yml"),
+            FileTypes::Json => write!(f, "json"),
+            FileTypes::Md => write!(f, "md"),
+            FileTypes::Ts => write!(f, "ts"),
+            FileTypes::Js => write!(f, "js"),
+        }
+    }
+}
```
Comment (tusharmath) [inline tailcall-prettier/src/lib.rs:27]: Can use `strum_macros`

File: tailcall-prettier/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
+use std::fmt::{Display, Formatter};
+use std::io::Write;
+use std::process::{Command, Stdio};
+
+use anyhow::{anyhow, Result};
+
+enum FileTypes {
```
Comment (tusharmath) [inline tailcall-prettier/src/lib.rs]:
```suggestion
enum Parser {
```

File: tailcall-prettier/src/lib.rs (hunk 1 above)
Comment (tusharmath) [reply tailcall-prettier/src/lib.rs:7]:
```suggestion
pub fn format_with_prettier<T: AsRef<str>>(code: T, parser: Parser) -> Result<String> {
```

File: tailcall-prettier/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+
+use std::io::Write;
+use std::process::{Command, Stdio};
+
+use anyhow::{anyhow, Result};
+
+pub fn format_with_prettier<T: AsRef<str>>(code: T, file_ty: T) -> Result<String> {
+    let mut child = Command::new("prettier")
```
Comment (tusharmath) [inline tailcall-prettier/src/lib.rs:8]: Make sure it uses the local prettier version.

File: tailcall-prettier/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::format_with_prettier;
+
+    #[test]
+    fn test_js() -> anyhow::Result<()> {
+        let prettier = format_with_prettier("const x={a:3};", "file.ts")?;
+        insta::assert_snapshot!(prettier);
```
Comment (tusharmath) [inline tailcall-prettier/src/lib.rs:34]: Snapshot isn't required for this test case.

Comment (ssddOnTop) [reply tailcall-prettier/src/lib.rs:34]: https://github.com/tailcallhq/tailcall/pull/1731/files#diff-acdc200f7c49c1de312eb8116bd8636eb7982395ff8c7f5b923cb1fbe8ffd102 already have that

**Pull request merged**